Fix pre-publish modal styles

octaedro opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR fixes a style error in the pre-publish panel. Before this PR the panel looked bad in small screens.

Before


![Screenshot 2024-02-12 at 14 36 49](https://github.com/woocommerce/woocommerce/assets/1314156/4bc359c4-5172-4cf0-9464-6013586326fa)

After

![Screenshot 2024-02-12 at 15 30 45](https://github.com/woocommerce/woocommerce/assets/1314156/3ae372ed-ede1-4ead-ae16-3466f073d909)

Closes #44560.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the New product editor and the Prepublish sidebar (product-pre-publish-modal) are enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. On a small screen, go to Products > Add New.
3. Add a product name and press `Publish`.
4. The panel should look good.
5. Verify that the panel looks good for different screen sizes.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

octaedro committed:
Fix pre-publish modal styles

octaedro committed:
Improve styles

octaedro committed:
Add changelog

retrofox commented on packages/js/product-editor/src/components/prepublish-panel/style.scss:
```diff
@@ -1,14 +1,17 @@
-		width: $sidebar-width + $border-width + 6px;
-		transform: translateX(+100%);
-		animation: product-publish-panel__slide-in-animation 0.1s forwards;
+		left: 72%;
+		width: 27%;
```
maybe you can use a var for this

retrofox reviewed (approved):
LGTM. I left a minor suggestion.

octaedro committed:
Add vars to styles

# Conflicts:
#	packages/js/product-editor/src/components/prepublish-panel/style.scss

octaedro committed:
Fix sidebar styles

octaedro committed:
Add changelog

octaedro commented:
Thank you @retrofox for your review. After rebasing the PR, I had to refactor it a little bit.
Could you take another look at this PR?

octaedro committed:
add overflow auto

octaedro committed:
Fix quotes

retrofox reviewed (approved):
LGTM ✅
